examples/mxnet/tree_lstm/train.py

```diff
@@ -0,0 +1,208 @@
+                                       sha1_hash='8084fbacc2dee3b1fd1ca4cc534cbfff3519ed0d')
+        with zipfile.ZipFile(zip_path, 'r') as zf:
+            zf.extractall()
+        if not data.utils.check_sha1('glove.840B.300d.txt',
+                                     sha1_hash='294b9f37fa64cce31f9ebb409c266fc379527708'):
+            warnings.warn('The downloaded glove embedding file checksum mismatch. File content '
+                          'may be corrupted.')
+
+def main(args):
+    np.random.seed(args.seed)
+    mx.random.seed(args.seed)
+
+    best_epoch = -1
+    best_dev_acc = 0
+
+    cuda = args.gpu >= 0
+    if cuda:
+        if args.gpu in mx.test_utils.list_gpus():
+            ctx = mx.gpu(args.gpu)
+        else:
+            print('Requested GPU id {} was not found. Defaulting to CPU implementation'.format(args.gpu))
+            ctx = mx.cpu()
+
+    if args.use_glove:
+        prepare_glove()
+
+    trainset = data.SST()
+    train_loader = gluon.data.DataLoader(dataset=trainset,
+                                         batch_size=args.batch_size,
+                                         batchify_fn=batcher(ctx),
+                                         shuffle=True,
+                                         num_workers=0)
+    devset = data.SST(mode='dev')
+    dev_loader = gluon.data.DataLoader(dataset=devset,
+                                       batch_size=100,
+                                       batchify_fn=batcher(ctx),
+                                       shuffle=True,
+                                       num_workers=0)
+
+    testset = data.SST(mode='test')
+    test_loader = gluon.data.DataLoader(dataset=testset,
+                                        batch_size=100,
+                                        batchify_fn=batcher(ctx),
+                                        shuffle=False, num_workers=0)
+
+    model = TreeLSTM(trainset.num_vocabs,
+                     args.x_size,
+                     args.h_size,
+                     trainset.num_classes,
+                     args.dropout,
+                     cell_type='childsum' if args.child_sum else 'nary',
+                     pretrained_emb = trainset.pretrained_emb,
+                     ctx=ctx)
+    print(model)
+    params_ex_emb =[x for x in model.collect_params().values()
+                    if x.grad_req != 'null' and x.shape[0] != trainset.num_vocabs]
+    params_emb = list(model.embedding.collect_params().values())
+    for p in params_emb:
+        p.lr_mult = 0.1
+
+    model.initialize(mx.init.Xavier(magnitude=1), ctx=ctx)
+    model.hybridize()
+    trainer = gluon.Trainer(model.collect_params('^(?!embedding).*$'), 'adagrad',
+                            {'learning_rate': args.lr, 'wd': args.weight_decay})
+    trainer_emb = gluon.Trainer(model.collect_params('^embedding.*$'), 'adagrad',
+                                {'learning_rate': args.lr})
+
+    dur = []
+    L = gluon.loss.SoftmaxCrossEntropyLoss(axis=1)
```

Comment:
In DyNet implementation, they use reduction=`sum` instead of `mean`, I'm also not sure which one is better, but in practice using `sum` produce higher results.